src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadataFactoryImplementation.java

```diff
@@ -0,0 +1,87 @@
+package com.exasol.adapter.dynamodb.dynamodbmetadata;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.model.GlobalSecondaryIndexDescription;
+import com.amazonaws.services.dynamodbv2.model.KeySchemaElement;
+import com.amazonaws.services.dynamodbv2.model.LocalSecondaryIndexDescription;
+import com.amazonaws.services.dynamodbv2.model.TableDescription;
+
+/**
+ * This class builds {@link DynamodbTableMetadata} by fetching the required information using a {@code describeTable}
```

Comment:
```suggestion
 * This class builds {@link DynamodbTableMetadataFactory} by fetching the required information using a {@code describeTable}
```

Reply:
Or `DynamodbTableMetadataFactoryImplementation`.

Reply:
no, this is the factory for `DynamodbTableMetadata`. `DynamodbTableMetadataFactory` is  the interface for the factory and `DynamodbTableMetadataFactoryImplementation` the implementation. Should I change the names somehow?

Reply:
Ahh okay. I get it now. It builds `DynamodbTableMetadata`. 
The all is okay from my side. But maybe you can remove the `Implementation` suffix and add `Base`, e.g, `BaseDynamodbTableFactory`. But it is up to you, please feel free to choose any naming.